lwt/happy_eyeballs_lwt.ml

```diff
@@ -152,21 +166,34 @@ let rec timer t =
   loop ()
 
 let ctr = ref 0
+let error_msgf fmt = Fmt.kstr (fun msg -> Error (`Msg msg)) fmt
 
-let create ?(happy_eyeballs = Happy_eyeballs.create (now ())) ?dns ?(timer_interval = Duration.of_ms 10) () =
-  let dns =
-    Option.value ~default:
-      (let timeout = Happy_eyeballs.resolve_timeout happy_eyeballs in
-       Dns_client_lwt.create ~timeout ())
-      dns
-  in
+let getaddrinfo record domain_name =
+  let open Lwt.Infix in
+  let getaddrinfo_option = match record with
+    | `A -> [ Unix.AI_FAMILY Unix.PF_INET ]
+    | `AAAA -> [ Unix.AI_FAMILY Unix.PF_INET6 ] in
+  let getaddrinfo_option = Unix.AI_SOCKTYPE Unix.SOCK_STREAM :: getaddrinfo_option in
+  Lwt.catch
+    (fun () -> Lwt_unix.getaddrinfo (Domain_name.to_string domain_name) "" getaddrinfo_option)
+    (fun exn -> Lwt.return []) >|= function
```

Comment:
```suggestion
    (fun _exn -> Lwt.return []) >|= function
```

Reply:
shouldn't we capture the error message, and report it to the user?

Reply:
Oh, yes, that's probably a better idea. I got a compiler error and didn't think much about it...